devicetypes/smartthings/cooper-receptacle.src/cooper-receptacle.groovy

```diff
@@ -0,0 +1,172 @@
+			state "default", label:'', action:"refresh.refresh", icon:"st.secondary.refresh"
+		}
+
+		main "switch"
+		details(["switch","refresh"])
+	}
+}
+
+def updated(){
+	def cmds = []
+	cmds << hubAction(zwave.configurationV1.configurationSet(configurationValue: [getPowerUpStateValue()],
+			parameterNumber: 5, size :1).format())
+	cmds << hubAction(zwave.protectionV1.protectionSet(protectionState: getProtection()).format())
+	cmds << hubAction(zwave.manufacturerSpecificV2.manufacturerSpecificGet().format())
+	sendHubCommand(cmds, 300)
+}
+
+def on() {
+	def cmd = []
+	cmd << hubAction(zwave.basicV1.basicSet(value: 0xFF).format())
+	cmd << hubAction(zwave.basicV1.basicGet().format())
+	sendHubCommand(cmd, 300)
+}
+
+def off() {
+	def cmd = []
+	cmd << hubAction(zwave.basicV1.basicSet(value: 0x00).format())
+	cmd << hubAction(zwave.basicV1.basicGet().format())
+	sendHubCommand(cmd, 300)
+}
+
+def refresh() {
+	def cmd = []
+	cmd << hubAction(zwave.basicV1.basicGet().format())
+	cmd << hubAction(zwave.configurationV1.configurationGet(parameterNumber: 5).format())
+	cmd << hubAction(zwave.protectionV1.protectionGet().format())
+	sendHubCommand(cmd, 300)
+}
+
+def parse(String description) {
+	def cmd = zwave.parse(description, [0x75:1])
+	if (cmd) {
+		zwaveEvent(cmd)
+	}
+	return []
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+	sendEvent(name: "switch", value: cmd.value ? "on" : "off")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicSet cmd) {
+	sendEvent(name: "switch", value: cmd.value ? "on" : "off")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.manufacturerspecificv2.ManufacturerSpecificReport cmd) {
+	if (cmd.manufacturerName) {
+		updateDataValue("manufacturer", cmd.manufacturerName)
+	}
+	if (cmd.productTypeId) {
+		updateDataValue("productTypeId", cmd.productTypeId.toString())
+	}
+	// productId 0 is a valid value
+	if (cmd.productId != null) {
+		updateDataValue("productId", cmd.productId.toString())
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd) {
+	//settings are read-only, so send event if settings are out of sync
+	if(cmd.parameterNumber == 5 && cmd.configurationValue[0] != getPowerUpStateValue()) {
+		sendEvent([descriptionText: "$device.displayName power up state settings are out of sync. Please save device preferences.",
+				   isStateChange: true])
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.protectionv1.ProtectionReport cmd) {
+	//settings are read-only, so send event if settings are out of sync
+	if(cmd.protectionState == getProtection()) {
+		sendEvent([descriptionText: "$device.displayName protection settings are out of sync. Please save device preferences.",
+				   isStateChange: true])
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	// Handles all Z-Wave commands we aren't interested in
+	log.debug "Unexpected zwave command $cmd"
+}
+
+private hubAction(formattedCommand) {
```

Comment:
If you're gonna do this already, I'd just pass in the unformatted command and call `format()` inside this method.